src/main/resources/jupyter/google_sign_in.js

```diff
@@ -18,7 +18,7 @@ var googleProject = $(googleProject);
 var clusterName = $(clusterName);
 
 // This is refreshed via postMessage from the client app.
-var googleClientId;
+var googleClientId = $(defaultClientId);
```

Comment:
Question: would this maintain backward compatibility with the post message way of doing this?

Reply:
Yup! In Firecloud, the postmessage route overwrites the googleClientId var in the receive method below